### PR TITLE
High low score methods

### PR DESCRIPTION
**Pull request by philjdelong**
highest_scoring_home_team
lowest_scoring_home_team
highest_scoring_away_team
lowest_scoring_away_team

### CODE REVIEW

File: lib/games_collection.rb [hunk 1]
```diff
@@ -85,4 +85,63 @@ def average_goals_per_season
...
+    home_team_goals.find do |team, goals|
+      goals == home_team_goals.values.sort.last
+    end.first
+  end
+
+  def lowest_scoring_home_team
+    home_team_goals.find do |team, goals|
+      goals == home_team_goals.values.sort.first
+    end.first
+  end
```
Comment (DanielEFrampton) [inline lib/games_collection.rb:139]: max_by could work great here, and need to chain fewer methods

File: test/games_collection_test.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 require_relative 'test_helper'
+require 'pry'
+require 'minitest/autorun'
+require 'minitest/pride'
```
Comment (DanielEFrampton) [inline test/games_collection_test.rb]: These shouldn't be needed since they're required by the test_helper

File: test/games_collection_test.rb [hunk 3]
```diff
@@ -106,4 +110,43 @@ def test_it_can_return_hash_of_average_goals_per_season
...
     assert_equal expected_hash, @games_collection.average_goals_per_season
   end
+
+  def test_it_can_tell_us_each_unique_home_team_id
+    #this validates that we're getting a condensed list of teams
+    assert_equal 17, @games_collection.home_teams.length
+  end
+
+  def test_it_can_tell_us_number_of_goals_per_home_team
+    assert_instance_of Hash, @games_collection.home_team_goals
```
Comment (DanielEFrampton) [inline test/games_collection_test.rb]: We chatted about this in person, but for the benefit of the team: yeah, I'd beef this test up a bit. Maybe also test the length of the hash, the data types of the keys and values, and other stuff you can test without writing out the whole hash.

File: test/games_collection_test.rb [hunk 4]
```diff
@@ -106,4 +110,43 @@ def test_it_can_return_hash_of_average_goals_per_season
...
+  end
+
+  def test_it_can_tell_us_each_unique_away_team_id
+
+    assert_equal 17, @games_collection.away_teams.length
+  end
+
+  def test_it_can_tell_us_number_of_goals_per_away_team
+
+    assert_instance_of Hash, @games_collection.away_team_goals
```
Comment (DanielEFrampton) [inline test/games_collection_test.rb:162]: See above comment on line 120.

**Review by DanielEFrampton — CHANGES_REQUESTED** [review]

**Review by raymondknguyen — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by DanielEFrampton — APPROVED** [review]

**Pull request merged**